packages/web3-utils/src/hash.ts

```diff
@@ -1,3 +1,218 @@
-// TODO: Implement later
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-export const sha3 = (data: string): string => data;
+import { keccak256 } from 'ethereum-cryptography/keccak';
+import { Numbers, typedObject, typedObject2, EncodingTypes } from './types';
+import { leftPad, rightPad, toTwosComplement } from './string_manipulation';
+import { utf8ToHex, hexToBytes, toNumber, bytesToHex } from './converters';
+import { isAddress, isHexStrict } from './validation';
+import {
+	InvalidStringError,
+	InvalidType,
+	InvalidBooleanError,
+	InvalidAddressError,
+	InvalidSizeError,
+	InvalidLargeValueError,
+	InvalidUnsignedIntegerError,
+	InvalidBytesError,
+} from './errors';
+
+const SHA3_NULL = '0xc5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470';
+
+/**
+ *
+ * computes the Keccak-256 hash of the string input and returns a hexstring
+ */
+export const sha3 = (data: string): string | null => {
```

Comment:
As per the usage of the code the input can be a lot of different type of values. 

```suggestion
export const sha3 = (data: Bytes): string | null => {
```